cylc/flow/platforms.py

```diff
@@ -187,7 +185,14 @@ def platform_from_name(platform_name=None, platforms=None):
     # later than site set platforms) to be matched first and override site
     # defined platforms.
     for platform_name_re in reversed(list(platforms)):
-        if re.fullmatch(platform_name_re, platform_name):
+        # We substitue commas with or without spaces to
+        # allow lists of platforms
+        if (
+            re.fullmatch(
+                re.sub(r',[^({\d+,\d+})]\s*', '|', platform_name_re),
```

Comment:
Not quite sure what this regex is doing? Think it's replacing `,` with `|` so `\s*,\s*`?

Reply:
That's exactly what it's doing, with an exception to prevent it splitting regexes:
i.e.:
```
platform_name_re = 'abcd\d{2,4}, other_platform' => 'abcd\d{2,4}|other_platform'
# not
'abcd\d{2|4}|other_platform'
```
It was the only use for a comma I could find in the regex spec.

Your suggestion is a good one.

```suggestion
                re.sub(r'\s*,[^({\d+,\d+})]\s*', '|', platform_name_re),
```

Reply:
Ah right didn't think about the match syntax.

Unfortunately this strips off the first character of the next platform name:

```python
>>> re.split(r'\s*,[^({\d+,\d+})]\s*', 'a,b')
['a', '']
```

Reply:
Might need some look aheads / behinds, this might be the right kind of pattern (untested):

```
\s*(?!{[\s\d]*),(?![\s\d]*})\s*
```

Reply:
Using a regex to parse regexes 😬

Reply:
What about a platform_re that's like `foo[^_,]`

Reply:
How very meta.

Reply:
I don't think commas make for valid hostnames so don't have to worry about that.

Reply:
```suggestion
\s*(?!{[\s\d]*),(?![\s\d]*})\s*
```
How _did_ you do _that_ (Write a working regex out of your head)

Reply:
Reviewing Matt's PRs for a few years, you learn so many trinkets that way.